src/app/components/turnos/gestor-agendas/pipes/botonesAgendaGeneral.pipe.ts

```diff
@@ -0,0 +1,116 @@
+
+        // Imprimir istado de carpetas en pdf
+        if (puedeImprimir && this.puedoImprimirCarpetas(agendas)) {
+            botones.imprimirCarpetas = true;
+        }
+        // Agregar una nota relacionada a la Agenda
+        if (puedeNota) {
+            botones.nota = true;
+        }
+        // Imprimir listado de turnos en pdf
+        if (puedeImprimir) {
+            botones.imprimir = true;
+        }
+        // Se pueden cambiar a estado pausada todas las agendas que no estén en estado planificacion
+        if (puedePausar && this.puedoPausar(agendas)) {
+            botones.pausar = true;
+        }
+        // Se pueden publicar todas las agendas que estén en estado planificacion, o si estado disponible y no tiene *sólo* turnos reservados
+        if (puedeHabilitar && this.puedoPublicar(agendas) && this.haySoloTurnosReservados(agendas)) {
+            botones.publicar = true;
+        }
+        // Se pueden pasar a disponible cualquier agenda en estado planificacion
+        if (puedeHabilitar && this.puedoDisponer(agendas)) {
+            botones.disponible = true;
+        }
+        // Se pueden borrar cualquier agenda en estado de planificacion
+        if (puedeBorrar && this.puedoBorrar(agendas)) {
+            botones.borrar = true;
+        }
+        // Se pueden reanudar las agendas en estado pausada
+        if (puedeReanudar && this.puedoReanudar(agendas)) {
+            botones.reanudar = true;
+        }
+
+        return botones;
+
+    }
+
+    puedoImprimirCarpetas(agendas) {
+        return !agendas.some((agenda: any) => agenda.estado === 'pendienteAsistencia' || agenda.estado === 'pendienteAuditoria' || agenda.estado === 'auditada' || agenda.estado === 'pausada' || agenda.estado === 'suspendida');
+    }
+
+    puedoPausar(agendas) {
+        return agendas.filter((agenda: any) => agenda.estado === 'planificacion' || agenda.estado === 'pausada' || agenda.estado === 'suspendida' || agenda.estado === 'auditada' || agenda.estado === 'pendienteAuditoria').length <= 0;
+    }
+
+    puedoPublicar(agendas) {
+        return agendas.filter((agenda: any) => agenda.estado !== 'planificacion' && agenda.estado !== 'disponible').length <= 0;
+    }
+
+    // Verifica que las agendas seleccionadas tengan al menos un turno de acceso directo poder para publicar la agenda
+    haySoloTurnosReservados(agendas) {
+        let band = false;
+        for (let x = 0; x < agendas.length; x++) {
+            for (let y = 0; y < agendas[x].bloques.length; y++) {
+                if (agendas[x].bloques[y].accesoDirectoProgramado === 0 &&
+                    agendas[x].bloques[y].accesoDirectoDelDia === 0 &&
+                    !agendas[x].dinamica) {
+                    band = false || band;
+                } else {
+                    band = true;
+                }
+            }
+        }
+        if (agendas.length) {
+            return band;
+        } else {
+            return false;
+        }
+    }
```

Comment:
Se podria simplificar, en caso de que quede mas entendible, con algo similar a..
```suggestion
    agendas.every(agenda => !agenda.dinamica && agenda.bloques.every(bloque => /* condiciones */ ));
```

Reply:
Corregido